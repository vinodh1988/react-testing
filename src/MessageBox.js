import React from 'react';

const MessageBox=({message,updateMessage})=>{
    console.log("Message Rendering...!!!");
    return(
        <div style={{backgroundColor:"lightgreen", 
        height:"300px",width:"600px",fontFamily:"Arial",padding: "10px",textAlign:"center", margin: "10px"}}>
             <h3>{message}</h3>
              <br/><br/>
             <button onClick={updateMessage}>Get New Message</button>
        </div>
    )
}


export default MessageBox;