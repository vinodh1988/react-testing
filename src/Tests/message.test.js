import {shallow,mount} from 'enzyme';
import React from 'react';
import MessageBox from '../MessageBox';

describe("Message component testing",function(){
let wrapper;
    beforeEach(()=>{
       wrapper=shallow(<MessageBox message="World is big"/>);
    })

  it("test should work",function(){
     expect(wrapper).toMatchSnapshot();
  })

  it("Should have one h3",function(){
      expect(wrapper.find('h3').length).toBe(1);
  })

  it("should have World is big in h3",function(){
      expect(wrapper.find('h3').text()).toBe('World is big');
  })

  it("should Mount properly and have World is big inh3",function(){
       wrapper=mount(<MessageBox message="World is big"/>);
       expect(wrapper.find('h3').text()).toBe('World is big');
   

  })

})