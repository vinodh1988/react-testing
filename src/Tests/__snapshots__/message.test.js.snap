// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component testing test should work 1`] = `
<div
  style={
    Object {
      "backgroundColor": "lightgreen",
      "fontFamily": "Arial",
      "height": "300px",
      "margin": "10px",
      "padding": "10px",
      "textAlign": "center",
      "width": "600px",
    }
  }
>
  <h3>
    World is big
  </h3>
  <br />
  <br />
  <button>
    Get New Message
  </button>
</div>
`;
